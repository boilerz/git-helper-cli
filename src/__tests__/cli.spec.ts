import updateNotifier from 'update-notifier';

import pkg from '../../package.json';
import run from '../cli';

describe('cli', () => {
  beforeEach(() => {
    jest.spyOn(process, 'exit').mockImplementation();
  });

  it('should setup cli', () => {
    const errorSpy = jest.spyOn(console, 'error');

    run([]);

    expect(errorSpy.mock.calls).toMatchSnapshot();
  });

  it('should call the update notifier', () => {
    run([]);

    expect(updateNotifier).toHaveBeenCalled();
    expect(updateNotifier({ pkg }).notify).toHaveBeenCalled();
  });
});
