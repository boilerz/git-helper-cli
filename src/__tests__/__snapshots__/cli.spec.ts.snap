// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli should setup cli 1`] = `
Array [
  Array [
    "git-helper <command>

Commands:
  git-helper preferences        Set preferences                     [aliases: p]
  git-helper yo <name> [color]  Say yo                          [aliases: hello]

Options:
  --version  Show version number                                       [boolean]
  --help     Show help                                                 [boolean]",
  ],
  Array [],
  Array [
    "Not enough non-option arguments: got 0, need at least 1",
  ],
]
`;
